modules/gdscript/gdscript.cpp

```diff
@@ -2167,17 +2176,33 @@ int GDScriptLanguage::profiling_get_accumulated_data(ProfilingInfo *p_info_arr,
 
 	MutexLock lock(this->mutex);
 
+	profiling_collate_native_call_data(true);
 	SelfList<GDScriptFunction> *elem = function_list.first();
 	while (elem) {
 		if (current >= p_info_max) {
 			break;
 		}
+		int last_non_internal = current;
 		p_info_arr[current].call_count = elem->self()->profile.call_count.get();
 		p_info_arr[current].self_time = elem->self()->profile.self_time.get();
 		p_info_arr[current].total_time = elem->self()->profile.total_time.get();
 		p_info_arr[current].signature = elem->self()->profile.signature;
-		elem = elem->next();
 		current++;
+
+		int nat_time = 0;
+		HashMap<String, GDScriptFunction::Profile::NativeProfile>::ConstIterator nat_calls;
+		nat_calls = elem->self()->profile.native_calls.begin();
+		while (nat_calls != elem->self()->profile.native_calls.end()) {
```

Comment:
```suggestion
		while (nat_calls) {
```
It has a boolean check, so this is simpler, the end iterator is just `Iterator(nullptr)`